### PR TITLE
Selection range

### PR DESCRIPTION
**Pull request by jasonlyu123**
#556 
The selection range seems to support multiple cursors, so the execution should be done in an `array.map` manner. Because of that, I didn't make the `PluginHost` implementing the same interface. Haven done the typescript part. Probably need to done a similar filtering like `getDocumentSymbols`

### CODE REVIEW

File: packages/language-server/src/plugins/svelte/features/getSelectionRanges.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    const { ast: { html } } = await svelteDoc.getCompiled();
+    const transpiled = await svelteDoc.getTranspiled();
+    const content = transpiled.getText();
+    const offset = svelteDoc.offsetAt(position);
+
+
+    const embedded = [script, style, moduleScript];
+    for (const info of embedded) {
+        if (isInTag(position, info)) {
+            // let other plugin done it
```
Comment (dummdidumm) [inline packages/language-server/src/plugins/svelte/features/getSelectionRanges.ts:26]:
```suggestion
            // let other plugins do it
```

File: packages/language-server/src/plugins/PluginHost.ts [hunk 2]
```diff
@@ -334,6 +335,32 @@ export class PluginHost implements LSProvider, OnWatchFileChanges {
         );
     }
 
+    async getSelectionRanges(
```
Comment (dummdidumm) [inline packages/language-server/src/plugins/PluginHost.ts:338]: Could you add a small comment above the function briefly explaining why this is a special case and the ranges have to be accumulated like this?

File: packages/language-server/src/plugins/svelte/features/getSelectionRanges.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+export async function getSelectionRange(
+    svelteDoc: SvelteDocument,
+    position: Position,
+) {
+    const { script, style, moduleScript } = svelteDoc;
+    const { ast: { html } } = await svelteDoc.getCompiled();
+    const transpiled = await svelteDoc.getTranspiled();
+    const content = transpiled.getText();
+    const offset = svelteDoc.offsetAt(position);
```
Comment (dummdidumm) [inline packages/language-server/src/plugins/svelte/features/getSelectionRanges.ts:20]: The offset here is wrong. The position has to be converted to a generated position first. For that you need to enhance the mappers inside `SvelteDocument` to support `getGeneratedPosition`.

**Review by dummdidumm — CHANGES_REQUESTED** [review]
Looks good to me. There is one bug though in `getSelectionRange` of the `SveltePlugin`: The position has to be mapped to the generated position first.

**Comment by dummdidumm** [comment]
Should I do another review or are there some things missing you want to do first?

**Comment by jasonlyu123** [comment]
Ok! It's ready for review again.

**Review by dummdidumm — APPROVED** [review]
👍

**Pull request merged**